server/zanata-war/src/test/java/org/zanata/adapter/SubtitleAdapterTest.java

```diff
@@ -0,0 +1,158 @@
+ */
+package org.zanata.adapter;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import org.junit.Before;
+import org.junit.Test;
+import org.zanata.common.LocaleId;
+import org.zanata.rest.dto.DTOUtil;
+import org.zanata.rest.dto.resource.Resource;
+
+import java.io.File;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author djansen <a href="mailto:djansen@redhat.com">djansen@redhat.com</a> on 6/02/17.
+ * @// TODO: 6/02/17 test ids
+ */
+public class SubtitleAdapterTest {
+
+    private SubtitleAdapter adapter;
+    private final String[] fileTypes = new String[]{"srt", "vtt", "sbt", "sub"};
+
+    @Before
+    public void setup() {
+        adapter = new SubtitleAdapter();
+    }
+
+    private Resource setupTestFile(String filename) {
+        String resPath = "src/test/resources/org/zanata/adapter/";
+        File testFile = new File(resPath.concat(filename));
+        assert testFile.exists();
+        Resource resource =
+                adapter.parseDocumentFile(testFile.toURI(), LocaleId.EN,
+                        Optional.absent());
+        //System.out.println(DTOUtil.toXML(resource));
+        return resource;
+    }
+
+    @Test
+    public void parseBasicSRT() {
+        Resource resource = setupTestFile("test-srt.srt");
+        assertThat(resource.getTextFlows()).hasSize(3);
+        assertThat(resource.getTextFlows().get(0).getContents()).isEqualTo(
+                ImmutableList.of("Line One"));
+    }
+
+    /*
+     * Entries that are of identical content must be individual textflows
+     */
+    @Test
+    public void testSimilarEntriesAreIndividual() {
+        String testText = "Exactly the same text";
+        Resource resource = setupTestFile("test-srt-duplicated.srt");
+        assertThat(resource.getTextFlows()).hasSize(2);
+        assertThat(resource.getTextFlows().get(0).getId()).isNotEqualTo(
+                resource.getTextFlows().get(1).getId());
+        assertThat(resource.getTextFlows().get(0).getContents()).isEqualTo(
+                ImmutableList.of(testText));
+        assertThat(resource.getTextFlows().get(1).getContents()).isEqualTo(
+                ImmutableList.of(testText));
+    }
+
+    /*
+     * WebVtt entries may start with a label or number, which can be safely ignored
+     */
+    @Test
+    public void testLabelledVtt() {
+        String testText = "Test subtitle 1";
+        Resource resource = setupTestFile("test-vtt-labelled.vtt");
+        assertThat(resource.getTextFlows()).hasSize(2);
+        assertThat(resource.getTextFlows().get(0).getContents()).isEqualTo(
+                ImmutableList.of(testText));
+    }
+
+    /*
+     * Entries may contain multiple lines, and be preserved as such
+     */
+    @Test
+    public void testMultilineSubtitleFile() {
+        String testText = "Test subtitle 1\nTest subtitle 1 line 2";
+        String targetFile = "test-$-multiline.$";
```

Comment:
Yes, that looks pretty good, but instead of having the various test classes modify the static variable `adapter`, it would be a bit cleaner to give them all a common (static) superclass, which would hold `adapter`, `parseTestFile()` and `setup()`.